File: muxer-commons/src/main/java/pt/um/tf/taskmux/commons/error/DuplicateException.java
package pt.um.tf.taskmux.commons.error;

import io.atomix.catalyst.buffer.BufferInput;
import io.atomix.catalyst.buffer.BufferOutput;
import io.atomix.catalyst.serializer.CatalystSerializable;
import io.atomix.catalyst.serializer.Serializer;

public class DuplicateException extends Exception implements CatalystSerializable {
    @Override
    public String getMessage() {
        return "Duplicate task present, retry get";
    }

    @Override
    public void writeObject(BufferOutput<?> buffer, Serializer serializer) {
    }

    @Override
    public void readObject(BufferInput<?> buffer, Serializer serializer) {}

}

File: muxer-commons/src/main/kotlin/pt/um/tf/taskmux/commons/error/DuplicateException.kt
package pt.um.tf.taskmux.commons.error

import io.atomix.catalyst.buffer.BufferInput
import io.atomix.catalyst.buffer.BufferOutput
import io.atomix.catalyst.serializer.CatalystSerializable
import io.atomix.catalyst.serializer.Serializer

class DuplicateException : Exception(), CatalystSerializable {
    override val message: String?
        get() = "Duplicate task present, retry get"


    override fun writeObject(buffer: BufferOutput<*>, serializer: Serializer) {}

    override fun readObject(buffer: BufferInput<*>, serializer : Serializer) {}

}
